File: platform-core-service/platform-core-service.Common/Models/DTOs/EntityDTO/Post/AdminPostDTO.cs
using platform_core_service.Common.Entities.BaseEntity;
using platform_core_service.Common.Utils.Enums;

namespace platform_core_service.Common.Models.DTOs.EntityDTO.Post
{
    public class AdminPostDTO : IBaseKey<string>
    {
        public string Id { get; set; } = null!;
        public string Title { get; set; } = null!;

        public string ContentPreview { get; set; } = null!;

        public string Slug { get; set; } = null!;
        public PostType PostType { get; set; }
        public string AuthorId { get; set; } = null!;
        public string? AuthorName { get; set; }

        public ModerationStatus ModerationStatus { get; set; }

        public int UpvoteCount { get; set; }
        public int DownvoteCount { get; set; }
        public DateTimeOffset? DateModified { get; set; }

        public DateTimeOffset? DateCreated { get; set; }
    }
}

File: frontend/src/types/admin/admin-post-dto.ts
export type PostType = 'MarkDown' | 'WYSIWYG';
export type ModerationStatus = 'Pending' | 'Approved' | 'Flagged' | 'InReview';

export interface SelectPostAuthorDTO {
  id: string;
  fullName: string;
  avatarUrl?: string;
  techStacks: string[];
}

export interface AdminPostDTO {
  id: string;
  title: string;
  contentPreview: string;
  slug: string;
  // Backend may return a numeric enum (0=MarkDown, 1=WYSIWYG) or string
  postType: PostType | number;
  // EntityType distinguishes "Post" vs "QA Post"
  entityType: string;
  authorId: string;
  author: SelectPostAuthorDTO;
  // Backend may return a numeric enum (0=Pending, 1=Approved, 2=Flagged, 3=InReview) or string
  moderationStatus: ModerationStatus | number;
  upvoteCount: number;
  downvoteCount: number;
  dateModified?: string;
  dateCreated?: string;
}
